crates/subspace-farmer-components/src/reading.rs

```diff
@@ -261,6 +257,8 @@ pub fn recover_extended_record_chunks(
     erasure_coding: &ErasureCoding,
 ) -> Result<Box<[Scalar; Record::NUM_S_BUCKETS]>, ReadingError> {
     // Restore source record scalars
+    // TODO: Recover into `Box<[Scalar; Record::NUM_S_BUCKETS]>` or else conversion into `Box` below
+    //  might leak memory
```

Comment:
This looks like a rather important TODO. Is there a follow up PR to this?

Reply:
It is essentially an upstream https://github.com/sifraitech/rust-kzg improvement. It is not that critical and basically means that compiler allocator can allocate more memory than necessary. See https://blog.polybdenum.com/2024/01/17/identifying-the-collect-vec-memory-leak-footgun.html for example of such issue. I don't think it is a big deal here and didn't notice issues related to this, just noted something we can improve in the future.